oemof_b3/tools/data_processing.py

```diff
@@ -0,0 +1,121 @@
+import pandas as pd
+import numpy as np
+
+
+def check_consistency_timeindex(df, index, name):
+    """
+    This function assert that values of a column in a stacked DataFrame are same
+    for all time steps
+
+    Parameters
+    ----------
+    index : string
+        Index of values to be checked in the DataFrame
+    name : string
+        Descriptive name of the index
+
+    Returns
+    -------
+    value : string
+        Single value of the series of duplicates
+
+    """
+    if np.all(df[index].array == df[index].array[0]):
+        value = df[index].array[0]
+        if value is None:
+            raise TypeError(
+                f"Your provided data is missing a {name}."
+                f"Please make sure you pass the {name} with {index}."
+            )
+        else:
+            return value
+    else:
+        raise ValueError(
+            f"The {name} of your provided data doesn't match for all entries. "
+            f"Please make sure to pass the {name} with {index}."
+        )
+
+
+def stack_timeseries(df):
+    _df = df.copy()
+
+    # Assert that _df has a timeindex
+    if not isinstance(_df.index, pd.DatetimeIndex):
+        raise TypeError(
+            "Your data should have a time series as an index of the format "
+            "'%Y-%m-%d %H:%M:%S'."
+        )
+
+    # Assert that frequency match for all time steps
+    if pd.infer_freq(_df.index) is None:
+        raise TypeError(
+            "No frequency of your provided data could be detected."
+            "Please provide a DataFrame with a specific frequency (eg. 'H' or 'T')."
+        )
+    else:
```

Comment:
You don't need this "else".

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/47/commits/8b0ea2db7abcb99128185834960a17a2a75efb79.